### PR TITLE
Add option to remove NVG when adding flashlights

### LINKED ISSUE #471: Feature Request: Removing NVGs off of units when applying the "Add flashlight module".
**Is your feature request related to a problem?**
No.

**Solution you'd like:**
When you put the "Add flashlight module" on a unit or a group their night vision goggles should be automatically removed. There's no reason for the unit to use flashlight if they still possess NVG.

**Alternatives you've considered:**
You can remove the NVGs in "arsenal" but the process is very tedious because you have to do it for each individual unit if there's a big group.

### PR DESCRIPTION
**Pull request by neilzar**
**When merged this pull request will:**
- Closes #471 
- Extend the `Add flashlight module` with an option to remove NVGs when enabling flashlights.

### CODE REVIEW

File: addons/modules/functions/fnc_moduleToggleFlashlights.sqf [hunk 1]
```diff
@@ -83,6 +84,12 @@ if (!_isAttached) then {
                 [QEGVAR(common,addWeaponItem), [_x, _weapon, selectRandom _flashlights], _x] call CBA_fnc_targetEvent;
             };
 
+            _nvg = assignedItems _x select 5;
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleToggleFlashlights.sqf:87]: `assignedItems` does not guarantee that NVGs are always at index 5.

File: addons/modules/stringtable.xml [hunk 2]
```diff
@@ -1724,6 +1724,9 @@
             <Polish>Dodaj wyposażenie</Polish>
             <Russian>Добавить снаряжение</Russian>
         </Key>
+        <Key ID="STR_ZEN_Modules_RemoveNVG">
+            <English>Remove NVG</English>
```
Comment (CreepPork) [inline addons/modules/stringtable.xml:1728]:
```suggestion
            <English>Remove NVGs</English>
```

**Review by CreepPork — CHANGES_REQUESTED** [review]

File: addons/modules/functions/fnc_moduleToggleFlashlights.sqf [hunk 3]
```diff
@@ -83,6 +84,12 @@ if (!_isAttached) then {
                 [QEGVAR(common,addWeaponItem), [_x, _weapon, selectRandom _flashlights], _x] call CBA_fnc_targetEvent;
             };
 
+            _nvg = hmd _x;
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleToggleFlashlights.sqf:87]: Missing `private` and could also be moved into `if` after `_removeNVG` is checked.

File: addons/modules/stringtable.xml [hunk 4]
```diff
@@ -1724,6 +1724,12 @@
             <Polish>Dodaj wyposażenie</Polish>
             <Russian>Добавить снаряжение</Russian>
         </Key>
+        <Key ID="STR_ZEN_Modules_RemoveNVG">
+            <English>Remove NVGs</English>
+        </Key>
+        <Key ID="STR_ZEN_Modules_RemoveNVG_Tooltip">
+            <English>Remove any Night Vision Goggles currently equiped.</English>
```
Comment (mharis001) [inline addons/modules/stringtable.xml:1731]: I think: "Remove any currently equipped Night Vision Goggles." reads a bit better. Also, equipped is misspelled.

File: addons/modules/functions/fnc_moduleToggleFlashlights.sqf [hunk 5]
```diff
@@ -83,6 +84,10 @@ if (!_isAttached) then {
                 [QEGVAR(common,addWeaponItem), [_x, _weapon, selectRandom _flashlights], _x] call CBA_fnc_targetEvent;
             };
 
+            if (_removeNVG && {hmd _x != ""}) then {
+                [QEGVAR(common,unlinkItem), [_x, _nvg], _x] call CBA_fnc_targetEvent;
+            };
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleToggleFlashlights.sqf:89]: This won't work now. I meant something like this:
```sqf
if (_removeNVG) then {
    private _nvg = hmd _x;

    if (_nvg != "") then {
        // ...
```

File: addons/common/XEH_postInit.sqf [hunk 6]
```diff
@@ -288,6 +288,11 @@
     _unit addweaponItem [_weapon, _item];
 }] call CBA_fnc_addEventHandler;
 
+[QGVAR(unlinkItem), {
+    params ["_unit", "_item"];
+    _unit unlinkItem _item;
+}] call CBA_fnc_addEventHandler;
```
Comment (mharis001) [inline addons/common/XEH_postInit.sqf:294]: `unlinkItem` is AG/EG so we don't need to use target events.

File: addons/modules/functions/fnc_moduleToggleFlashlights.sqf [hunk 7]
```diff
@@ -83,6 +84,14 @@ if (!_isAttached) then {
                 [QEGVAR(common,addWeaponItem), [_x, _weapon, selectRandom _flashlights], _x] call CBA_fnc_targetEvent;
             };
 
+            if (_removeNVG) then {
+                private _nvg = hmd _x;
+
+                if (_nvg != "") then {
+                    _unit unlinkItem _x;
```
Comment (mharis001) [inline addons/modules/functions/fnc_moduleToggleFlashlights.sqf:91]: Should be `_nvg` instead of `_x`.

**Review by mharis001 — APPROVED** [review]

**Review by CreepPork — APPROVED** [review]

**Pull request merged**